Sources/PluginLibrary/Array+Extensions.swift

```diff
@@ -0,0 +1,22 @@
+// Sources/PluginLibrary/Array+Extensions.swift - Additions to Arrays
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/master/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+
+import Foundation
+
+extension Array {
+  /// Like map, but calls the transform with the index and value.
+  func enumeratedMap<T>(_ transform: (Int, Element) throws -> T) rethrows -> [T] {
```

Comment:
Why not just use `yourArray.enumerated().map { ... }` at the call site instead? Your transform will have to change slightly because it'll take a *single* argument that is a tuple containing the index and element, instead of taking them as separate elements, but otherwise, it's the same.